src/components/LocationListView.js

```diff
@@ -87,17 +87,27 @@ export default class LocationListView extends Component {
 
     render() {
         const primary = this.props.primary;
-        if (this.props.rows.length === 0) {
+
+        if (this.props.rows.length == 0 && this.props.loaded) {
+            return (
+                <View style={styles.container}>
+                    <View style={styles.header}>
+                        <Text style={styles.headerText}>No locations found</Text>
```

Comment:
I know that you don't added this but we should add this string to messages file.